src/sidebar/util/collections.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Utility functions for collections: Sets, Maps, Arrays and Map-like objects
+ */
+
+/**
+ * Return the number of elements in `ary` for which `predicate` returns true.
+ *
+ * @param {Array} ary
+ * @param {Function} predicate
```

Comment:
```suggestion
 * @template T
 * @param {T[]} ary
 * @param {(item: T) => boolean} predicate
```

Minor suggestion that could potentially catch usage errors.